src/test/java/nextstep/subway/domain/SectionsTest.java

```diff
@@ -30,38 +32,111 @@ void setUp() {
         givenSections = new Sections(Lists.newArrayList(givenSection));
     }
 
-    /**
-     * When 역 제거 요청 시
-     * Then 제거가 된다
-     */
-    @DisplayName("역 제거 요청 시 구간이 제거가 된다")
-    @Test
-    void 역_제거_요청_시_구간이_제거가_된다() {
-        // When
-        givenSections.removeLastSection(양재역);
+    @Nested
+    @DisplayName("역 제거 요청")
+    class RemoveSection {
 
-        // Then
-        assertThat(givenSections.isSectionsEmpty()).isTrue();
-    }
+        /**
+         * When 마지막 역 제거 요청 시
+         * Then 제거가 된다
+         */
+        @DisplayName("마지막역 제거 요청 시 구간이 제거가 된다")
+        @Test
+        void 마지막역_제거_요청_시_구간이_제거가_된다() {
+            // Given
+            Station 정자역 = new Station("정자역");
+            givenSections.addSection(신분당선, 양재역, 정자역, 10);
+
+            // When
+            givenSections.removeSection(정자역);
+
+            // Then
+            assertThat(givenSections.getOrderedStations()).containsExactly(강남역, 양재역);
+        }
+
+        /**
+         * Given 노선에 새로운 구간 추가를 요청 하고
+         * When 지하철 노선의 어느 구간이든 제거를 요청 하면
+         * Then 노선에 구간이 제거된다
+         * Then 거리는 합산이 된다
+         */
+        @DisplayName("어느 구간이든 제거 할 수 있다")
+        @Test
+        void 지하철_노선에_어느_구간이든_제거_할_수_있다() {
+            // given
+            Station 정자역 = new Station("정자역");
+            givenSections.addSection(신분당선, 양재역, 정자역, 10);
+
+            // when
+            givenSections.removeSection(양재역);
+
+            // then
+            assertThat(givenSections.getOrderedStations()).containsExactly(강남역, 정자역);
+            Section 강남역_구간 = givenSections.findSectionByStation(강남역);
+            assertThat(강남역_구간.getDistance()).isEqualTo(20);
+            assertThat(강남역_구간.getUpStation()).isEqualTo(강남역);
+            assertThat(강남역_구간.getDownStation()).isEqualTo(정자역);
```

Comment:
다음과 같이 검증할 수 있을 것 같은데 어떻게 생각하시나요?
어떠한 차이가 있을까요? 😄 
```suggestion
            assertThat(강남역_구간).isEqualTo(new Section(신분당선, 강남역, 정자역 ,20));
```